### ISSUE TITLE
[Improvement] Image Token Speed

### ISSUE THREAD

**Issue by Veldkornet** [athombv/homey#2173]
**Metadata**

* **Homey Version**:
* **Homey Log Code**:
* **Smartphone Platform**:
* **Smartphone App Version**:

**If related to a Homey App:**
* **Homey App Version**: ImageGrabber 0.1.0
* **Homey App Log Code**:
* **Affected Devices (preferably product codes)**:

**What did you try to achieve?**
Use ImageGrabber to grab an image

**What did you expect as result?**
Image was grabbed and sent immediately

**What was the result?**
Image was grabbed, but was considerably slower than what was expected. 

**Any other remarks**

Is the new version perhaps slower than it was previously? Or is it just me?

Usually, when someone rang the doorbell, I got the image almost instantly. Now I get it up to 10 seconds later once there is no one at the door anymore. I have the impression that there is now a bit of a delay between Image Grabber’s “Grab Image” and “When an Image is grabbed”

The "grabbing" code is exactly identical. We now just use the "Athom" way of storing the images. Is this maybe a bit slower?

See https://github.com/jordenc/image.grabber/issues/4

**Comment by jeroenvollenbrock** [comment]
Image tokens only request the image when the token is actually used. This makes sure that only when a token is needed the image is fetched. Are there any delays in your flow?

**Comment by Veldkornet** [comment]
Just as a FYI, today it took over a minute for me to get the image when someone rang my doorbell, where it used to be almost instant. Using exactly the same flow as I always have.

**Comment by Fire69** [comment]
I'm having the same problem here.
First I thought it was caused by the Telegram app, but apparently it's the image grabbing itself that has a huge delay now...

**Comment by bvdbos** [comment]
Where is the delay? What happens if a second action is added to the "send image", something like "let Homey speak" or "color ledring"?

**Comment by Veldkornet** [comment]
Right, so to get things cleared up once and for all, I added some logs to the flow:

![image](https://user-images.githubusercontent.com/10297361/37289617-6b797680-260a-11e8-9ae0-6cf5a7ba4c51.png)
![image](https://user-images.githubusercontent.com/10297361/37289705-9e0210f8-260a-11e8-850a-c2e0dfebfd66.png)

So, assuming that these logs work how I expect them to, the flows are all done instantly. so... I give up....

Here are my logs from the above (note that it took a minute for me to get the image, where it used to be instant):

2018-03-12 15:27:08 Image Grab Flow Started
2018-03-12 15:27:08 Image Grabbed
2018-03-12 15:27:08 Send Image Flow Started.
2018-03-12 15:27:08 Image Sent

**Comment by Fire69** [comment]
I have 2 flows:
- When something happens, grab image (normally from my Foscam, used another image from the net as test)
- When an image is grabbed, send it to Telegram

I added a notification to both flows (send Test1 / Test2).
When I start the first flow with the test button, I almost instantly get 2 notifications (Test1/Test2), but I get no image in Telegram.
On the Grab Image card, the circle keeps spinning
![image](https://user-images.githubusercontent.com/1757287/37289993-6e669638-260b-11e8-959a-582f26aff7ad.png)

**Comment by Veldkornet** [comment]
Maybe @jordenc can help us with some insight here

**Comment by Fire69** [comment]
The images not arriving is intermittent actually...
Normally it's an http from my ipcam.
My actual flow is in fact grabbing 4 images with a delay of 1sec between each grab.
Just tested again, first image arrived after ~15sec, then it took another 30sec before the other 3 all arrived at the same time.

Even after getting the images, the circles keep spinning
![image](https://user-images.githubusercontent.com/1757287/37290845-9aba098e-260d-11e8-8d63-088e2eab503e.png)

**Comment by sossienl** [comment]
Same issue here. Homey seems to freeze for a while as well, becoming fully unresponsive for a while.

**Comment by Veldkornet** [comment]
Okay, agreed, it seems as though it is the actual grabbing which is slow. Appears to be. In fact, it just spins indefinitely.

So who can fix it?

**Comment by Veldkornet** [comment]
FYI, I think this never ending spinning wheel may also be (one of) the causes for #2381

**Comment by 2benl** [comment]
I have the same problem. When I was using Synology and Telegram with it's old own image tag system the pictures were almost instant. Since Synology stopt working I use image grabber with Homey's own image tokens the time is about a minute. It looks to me that in the new system the size of the image is a much bigger item.

**Comment by WeeJeWel** [comment]
A fix for the speed is planned for Homey v2.0.0
